src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -165,6 +167,7 @@ StartMetadataSyncToNode(const char *nodeNameString, int32 nodePort)
 		return;
 	}
 
+	UpdateHasmetadataOnWorkersWithMetadata(nodeNameString, nodePort, "true");
```

Comment:
Can we use `SetWorkerColumn()` for both `hasMetadata` and `metadataSynced` columns? Now that we have multiple ways of doing similar things, I think if we can converge into a single implementation, it'd improve the code.

Also, as I implicitly mentioned above, I think we should also sync `metadatasynced` filed, otherwise we'd have a little inconsistent metadata on the workers:

```SQL
select nodeport, hasmetadata, metadatasynced FROM pg_dist_node;
 nodeport | hasmetadata | metadatasynced 
----------+-------------+----------------
     9700 | t           | f
```